### DISCUSSION TITLE
Consumindo serviço externo

### DISCUSSION THREAD

**Discussion by igorRomulo** [balta-io/7196#46] [Perguntas e Respostas] [answered]
Balta vamos supor que preciso validar um CEP pelo correios, nessa estrutura de projeto onde eu colocaria o serviço externo que teria que consumir, cria uma pasta de service no domain e criaria uma interface e depois o restante fica na infra?

**Comment by andrebaltieri** [comment, marked as answer]
Show, é uma ótima pergunta... Onde você colocaria e por quê?
Me conta aí que evoluímos essa conversa...

**Reply by igorRomulo** [reply]
dentro do **Dominion** > um command, um handler e uma interface para o serviço, dentro da **Infra** a class de serviço com a interface de serviço

**Reply by andrebaltieri** [reply]
Perfeito! Eu faria exatamente assim, criando core e interfaces no domínio e a implementação na infra!
